### PR TITLE
analytics: stop sending analytics in tests

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: Library/Homebrew/test/utils/analytics_spec.rb [hunk 1]
```diff
@@ -117,9 +117,8 @@
       ENV.delete("HOMEBREW_NO_ANALYTICS")
       ENV["HOMEBREW_ANALYTICS_DEBUG"] = "true"
       ENV["HOMEBREW_ANALYTICS_ENABLE_INFLUX"] = "true"
-      expect do
-        described_class.report_influx(:install, action, true, developer: true, CI: true)
-      end.to output(/--data-raw install,[a-zA-Z=,]*,action=formula_name\\ --head/).to_stdout
+      expect(described_class).to receive(:defered_curl).once
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/utils/analytics_spec.rb:120]:
```suggestion
      expect(described_class).to receive(:deferred_curl).once
```

**Review by MikeMcQuaid — APPROVED** [review]
One typo, good to ship after than that.

File: Library/Homebrew/utils/analytics.rb [hunk 2]
```diff
@@ -99,8 +99,13 @@ def report_influx(category, action, on_request, additional_tags = {})
           "--data-raw", "#{category},#{tags} count=1i #{Time.now.to_i}"
         ]
 
-        curl = Utils::Curl.curl_executable
         url = "https://#{INFLUX_HOST}/api/v2/write?bucket=#{INFLUX_BUCKET}&precision=s"
+        defered_curl(url, args)
```
Comment (EricFromCanada) [inline Library/Homebrew/utils/analytics.rb:103]:
```suggestion
        deferred_curl(url, args)
```

File: Library/Homebrew/utils/analytics.rb [hunk 3]
```diff
@@ -99,8 +99,13 @@ def report_influx(category, action, on_request, additional_tags = {})
...
+      end
+
+      sig { params(url: String, args: T::Array[String]).void }
+      def defered_curl(url, args)
```
Comment (EricFromCanada) [inline Library/Homebrew/utils/analytics.rb:107]:
```suggestion
      def deferred_curl(url, args)
```

**Review by EricFromCanada — CHANGES_REQUESTED** [review]

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**